remote_execution/oss/re_grpc/src/client.rs

```diff
@@ -488,6 +493,7 @@ impl REClient {
             grpc_clients,
             capabilities,
             instance_name,
+            find_missing_cache: Mutex::new(LruCache::new(NonZeroUsize::new(50 << 20).unwrap())), // 50Mb
```

Comment:
Since `TDigest`s are small, this is actually really big. That means that it's going to have potential problems with artifact expiration. Can we add some kind of a timestamp to this and then invalidate this cache after, say, 12 hours so that we know we won't have TTL problems?